File: python/distinct_count.py
class Solution():
    def distinct(self,arr,no):
        count = 0
        isDistinct = True

        for i in range(0,no):
            isDistinct = True
            for j in range(0,i):
                if arr[i] == arr[j]:
                    isDistinct = False
                    break
            if isDistinct == True:
                count += 1
        
        return count

if __name__ == '__main__':
    no = int(input())
    array = [int(x) for x in input().split()]
    obj = Solution()
    print(obj.distinct(array,no))

File: cpp/distinct_count.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution
{
    public:
        int distinct(int arr[],int no)
        {
            int count = 0;
            bool isDistinct = true;

            for(int i = 0;i<no;i++)
            {
                isDistinct = true;

                for(int j = i-1;j>=0;j--)
                {
                    if(arr[i] == arr[j])
                    {
                        isDistinct = false;
                        break;
                    }
                }
                if(isDistinct == true)
                    count++;
            }
            return count;
        }
};
int main()
{
    int no;
    cin>>no;
    int array[no];
    for(int i = 0;i<no;i++)
    {
        cin>>array[i];
    }
    Solution obj;
    cout<<obj.distinct(array,no)<<endl;
    return 0;    
}